### PR TITLE
Display more informative error message when solve() fails

### LINKED ISSUE #64: More informative feedback when solver fails
When the solver fails, the error dialog just says, "Solver failed to return a calibration". More descriptive errors are printed in the terminal output, but there's often a lot of irrelevant information there that can make the actual error hard to find. I propose modifying the solver plugin interface to include an error string output, which could then be appended to the error dialog. I think this issue would be appropriate for a first-time contributor. I plan to have the good-first-issue stuff prepared in time for World MoveIt Day.

### PR DESCRIPTION
**Pull request by JStech**
Now `solve()` will output a helpful error message, and the GUI will display this in the dialog that pops up. Fixes #64 .
handeye_cal_solver_error

### CODE REVIEW

File: moveit_calibration_plugins/handeye_calibration_solver/include/moveit/handeye_calibration_solver/handeye_solver_base.h [hunk 1]
```diff
@@ -71,7 +72,7 @@ class HandEyeSolverBase
    * @param solver_name The algorithm used in the calculation.
    * @return If the calculation succeeds, return true. Otherwise, return false.
    */
-  virtual bool solve(const std::vector<Eigen::Isometry3d>& effector_wrt_world,
```
Comment (felixvd) [inline moveit_calibration_plugins/handeye_calibration_solver/include/moveit/handeye_calibration_solver/handeye_solver_base.h:74]: Let's add the new argument at the end and set a default value of "".

File: moveit_calibration_gui/handeye_calibration_rviz_plugin/src/handeye_control_widget.cpp [hunk 2]
```diff
@@ -404,7 +404,8 @@ bool ControlTabWidget::solveCameraRobotPose()
 {
   if (solver_ && !calibration_solver_->currentText().isEmpty())
   {
-    bool res = solver_->solve(effector_wrt_world_, object_wrt_sensor_, sensor_mount_type_,
+    std::string error_message("Unknown error");
```
Comment (felixvd) [inline moveit_calibration_gui/handeye_calibration_rviz_plugin/src/handeye_control_widget.cpp:407]: To me this line doesn't communicate well that this is only an initial value that's meant to be overwritten. 

I would either add a comment `// Default value`, or initialize it as empty and display `"Error: " + error_message` in the GUI dialog. I think I prefer the latter.

**Review by felixvd — APPROVED** [review]
LGTM after the few changes

File: moveit_calibration_plugins/handeye_calibration_solver/include/moveit/handeye_calibration_solver/handeye_solver_base.h (hunk 1 above)
Comment (JStech) [reply moveit_calibration_plugins/handeye_calibration_solver/include/moveit/handeye_calibration_solver/handeye_solver_base.h:74]: I read this suggestion and thought "Why didn't I do that in the first place?" and now CI reminds me--you can't have a non-const ref argument with a default value. So I did a pointer output parameter instead, which is what the [ROS C++ style guide](http://wiki.ros.org/CppStyleGuide#Output_arguments) recommends, although I don't see this approach used much elsewhere in MoveIt.

File: moveit_calibration_gui/handeye_calibration_rviz_plugin/src/handeye_control_widget.cpp (hunk 2 above)
Comment (JStech) [reply moveit_calibration_gui/handeye_calibration_rviz_plugin/src/handeye_control_widget.cpp]: Done

**Pull request merged**